src/content/1.7/development/components/grid/tutorials/modify-grid-in-module.md

```diff
@@ -3,4 +3,170 @@ title: How to modify an existing Grid in a module?
+
+```php
+/**
+ * Hooks allows to modify Logs grid definition.
+ * This hook is a right place to add/remove columns or actions (bulk, grid).
+ *
+ * @param array $params
+ */
+public function hookActionCustomerGridDefinitionModifier(array $params)
+{
+    /** @var GridDefinitionInterface $definition */
+    $definition = $params['definition'];
+
+    /** @var ColumnCollection */
+    $columns = $definition->getColumns();
+}
+```
+
+Once we have registered the hook, we can alter the columns:
+
+* remove the existing ones,
+* add new ones,
+* and position them precisely;
+
+```php
+/** @var ColumnCollection */
+$columns = $definition->getColumns();
+$columns->remove('social_title')
+    ->remove('active')
+    ->remove('optin')
+;
+```
+
+Afer the activation of the module, the grid should looks like this:
+
+![grid-tuto1](https://user-images.githubusercontent.com/1247388/52344104-33516780-2a1a-11e9-8f88-b299c6e444f7.png)
+
+It's not really nice... because we also need to delete the related search filters!
+
+Let's do that:
+
+```php
+/** @var FilterCollection $filters */
+$filters = $definition->getFilters();
+
+$filters->remove('social_title')
+    ->remove('active')
+    ->remove('optin')
+;
+```
+
+And this time, our grid is awesome!
+
+![grid-tuto-2](https://user-images.githubusercontent.com/1247388/52344105-33516780-2a1a-11e9-9ec8-96427561be15.png)
+
+## Re-ordering grid columns and filters
+
+When adding a new column, it's super easy to position it in a grid.
+
+The `ColumnCollection` class have two functions for that:
+
+* `addAfter($columnId, ColumnInterface $column)`
+* `addBefore($columnId, ColumnInterface $column)`
+
+If you need to insert a new column "$newColumn" after a specific column with the id "name", this is how you can do that:
+
+```php
+$newColumn = new DataColumn('exemple') ...
+$columns->addAfter('name', $newColumn);
+```
+
+The operation is **more difficult** with an existing column. We need to remove and add it again to the right position:
+
+```php
+$columns->remove('the_column_we_need_to_move');
+
+$columnWeNeedToMove = new ... // we create the column
+$columns->addBefore('sales', $ccolumnWeNeedToMove);
+```
+
+### Re-ordering filters?
+
+There is a good news here: filters are linked to the columns thanks to the function `setAssociatedColumn` of the `FilterInterface` contract.
+
+This means that you don't need to re-order filters when you re-order columns: great!
+
+If you don't define a new filter when creating a new column, you won't be able to search using this column as a filter.
+
+But... what if you want to make this column searchable as most of the existing ones?
+
+Also, you may wondering at this moment how PrestaShop is able to fill the column data?
+
+## Fill the column data and configure the filter
+
+Each grid is defined by a Definition and a Query Builder.
+
+The Query Builder(s) have two responsibilities:
```

Comment:
query builders only builds 2 queries: one for retrieving grid data and another for total count of data. 

but data providers (`DataFactory`) uses these queries (from query builders) to get data and build it that way so Grid component can understand.

Reply:
I know, but from a "noob/beginner" point of view, isn't enough to know that at this moment?

Reply:
well, we can explain actual query builder responsibilities (which is build querys) and say that these queries are then used internally (without diving into too much details) by Grid component to get data, wdyt?

Reply:
I've improved this part.